### PR TITLE
Add delete  message API function

### PR DESCRIPTION
**Pull request by shashi918**
Signed-off-by: shashi918 <sr0298037@gmail.com>

Added delete message API

### CODE REVIEW

File: src/gwbot/include/gwbot/lib/tg_api/delete_message.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#include <gwbot/lib/tg_api.h>
+
+
+typedef struct _tga_delete_chat_t {
+        int64_t		chat_id;
+	uint64_t	message_id;
+} tga_delete_chat_t;
+
+
+int tga_delete_chat(tga_handle_t *handle, const tga_delete_chat_t *ctx);
```
Comment (ammarfaizi2) [inline src/gwbot/include/gwbot/lib/tg_api/delete_message.h:22]:
```suggestion
int tga_delete_msg(tga_handle_t *handle, const tga_delete_msg_t *ctx);
```

**Comment by ammarfaizi2** [comment]
Please fix it based on the code review above.

File: src/gwbot/lib/tg_api/delete_message.c [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+			"chat_id=%" PRId64
+			"&message_id=%" PRIu64,
+			ctx->chat_id,
+			ctx->message_id);
+
+	tga_set_body(handle,buf);
+	tga_set_method(handle, "deleteMessage");
+	return tg_api_post(handle);
+}
+
```
Comment (ammarfaizi2) [inline src/gwbot/lib/tg_api/delete_message.c:21]: Remove this unnecessary whitespace

**Comment by ammarfaizi2** [comment]
Bad indentation and whitespace.
Don't use spaces for struct indentation. Use tabs.

**Comment by ammarfaizi2** [comment]
Remove unnecessary whitespace in delete_message.c line 21.

File: src/gwbot/include/gwbot/lib/tg_api/delete_message.h [hunk 3]
```diff
@@ -13,13 +13,13 @@
 #include <gwbot/lib/tg_api.h>
 
 
-typedef struct _tga_delete_chat_t {
+typedef struct _tga_delete_message_t {
         int64_t		chat_id;
-	uint64_t	message_id;
-} tga_delete_chat_t;
+        uint64_t        message_id;
```
[the remark this answers is not in the document]
Comment (shashi918) [reply src/gwbot/include/gwbot/lib/tg_api/delete_message.h:18]: image

here showing fine why?

File: src/gwbot/lib/tg_api/delete_message.c (hunk 2 above)
Comment (shashi918) [reply src/gwbot/lib/tg_api/delete_message.c:21]: ok fixing the white spaces  problem

File: src/gwbot/include/gwbot/lib/tg_api/delete_message.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (ammarfaizi2) [reply src/gwbot/include/gwbot/lib/tg_api/delete_message.h:18]: They are not tabs.

File: src/gwbot/lib/tg_api/delete_message.c [hunk 4]
```diff
@@ -15,7 +15,7 @@ int tga_delete_message(tga_handle_t *handle, const tga_delete_message_t *ctx)
 		ctx->message_id);
 
 	tga_set_body(handle,buf);
-	tga_set_method(handle, "deleteMessage");
+	tga_set_method(handle,"deleteMessage");
```
Comment (ammarfaizi2) [inline src/gwbot/lib/tg_api/delete_message.c:18]: Put space after comma

**Review by ammarfaizi2 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**